python/ray/tests/test_cancel.py

```diff
@@ -306,6 +306,23 @@ def many_resources():
     assert ray.get(many_fut, timeout=30)
 
 
+@pytest.mark.parametrize("use_force", [True, False])
+def test_cancel_allow_retry(ray_start_regular, use_force):
```

Comment:
how is this going to work if recursive=False.  Could you also add tests for that?

Reply:
Hmm I'm not really sure how `recursive=True` is implemented, but this should work out of the box since Ray will clean up child tasks of killed workers. So when the cancelled task re-executes, it will submit entirely new child tasks instead of retrying the old ones.

Can you test this behavior out with recursive=True and False, though? It would be good to know the current behavior.